src/util/evalUnits.ts

```diff
@@ -53,19 +52,37 @@ export enum EvalMessages {
  * the first is the actual calculated number (or null if none) while
  * the second is the message that should appear in case of error
  */
-export function evalUnits(input: string): [number | null, string] {
+export function evalUnits(input: string, chainDecimals: number): [BN | null, string] {
+  //sanitize input to remove + char if exists
+  input = input && input.replace('+', '');
   if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
     return [null, EvalMessages.GIBBERISH];
   }
+  // find the character from the alphanumerics
+  const symbol = input.replace(/[0-9.,]/g, '');
+  // find the value from the si list
+  const siVal = si.find((s) => s.symbol === symbol);
+  const numberStr = input.replace(symbol, '').replace(',', '.');
+  let numeric: BN = new BN(0);
+  if (siVal) {
+    let action: number | BN;
+    const containDecimal = input.replace(/[,]/g, '.').includes('.');
+    if (containDecimal) {
+      action = parseFloat(numberStr.replace(/[,]/g, '.')) * 10 ** chainDecimals;
+    } else {
+      action = new BN(numberStr).mul(new BN(10).pow(new BN(chainDecimals)));
+    }
+    numeric = new BN(action).mul(siVal.value);
+
+    // Temporary workaround so that UI will not break
+    if (numeric.bitLength() > 64) {
```

Comment:
Why is this needed? We do support chains with `128` bits of balance :)

Reply:
as said [here](https://matrix.to/#/!QDITCUQOfCSGTtPxkj:matrix.parity.io/$brl9NG87YDEdskWPyvSivDBrxPuq3Y4rc0hnpO_czPg?via=matrix.parity.io):

```
Unhandled Rejection (Error): createType(Call):: Call: failed decoding balances.transfer:: Struct: failed on args: {"dest":"LookupSource","value":"Compact<Balance>"}:: Struct: failed on value: Compact<Balance>:: Balance: Input too large. Found input with 80 bits, expected 64
```
but this time this happen at the point where polkadot API is called:
```
call = (await targetApi.tx.balances.transfer(receiverAddress, transferAmount || new BN(0))).toU8a();
```
based on numeric.bitLength() the MAX token shorthand we can get is G.
If API is not called (not select for example something from the dropdown) everything works fine.

Reply:
But this stuff depends on chain-specific parameters (like `Balance` type size and number of decimals). This restriction seems artificial.
I'm fine doing something like:
```
function isAmountSupported(parsedBalance: BN): bool {
  try {
      api.createType('Balance', parsedBalance);
      return true;
  } catch (e) {
      return false;
  }
```

to check if given balance is goign to be supported by the chain or not.